### PR TITLE
op-e2e: improve compat with force-ecotone testing

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

When using `OP_E2E_USE_ECOTONE` you can force all tests to use a deploy-config that configures ecotone activation at genesis. Some tests couldn't handle this very well, and needed exceptions / readjustments. This gets us closer to making ecotone a default in op-e2e at a later stage.

Depends on #9071

### CODE REVIEW

File: op-e2e/actions/l2_batcher_test.go [hunk 1]
```diff
@@ -50,6 +50,27 @@ func TestL2BatcherBatchType(t *testing.T) {
 	}
 }
 
+// applyDeltaTimeOffset adjusts fork configuration to not conflict with the delta overrides
+func applyDeltaTimeOffset(dp *e2eutils.DeployParams, deltaTimeOffset *hexutil.Uint64) {
```
Comment (tynes) [inline op-e2e/actions/l2_batcher_test.go:54]: This approach doesn't scale well but is fine for now

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Looks like this needs a rebase + conflict fix

**Review by ajsutton — APPROVED** [review]
Seems like some time soon we should introduce a upgrades helper type of thing to provide utility methods to correctly enable specific upgrades rather than having individual tests modify the config and then need to be updated as the default settings change or new upgrades are added.

File: op-node/rollup/derive/engine_consolidate.go [hunk 2]
```diff
@@ -47,8 +47,25 @@ func AttributesMatchBlock(rollupCfg *rollup.Config, attrs *eth.PayloadAttributes
...
 	}
-	if envelope.ParentBeaconBlockRoot != attrs.ParentBeaconBlockRoot {
-		return fmt.Errorf("parent beacon block root does not match. expected %v. got: %v", attrs.ParentBeaconBlockRoot, envelope.ParentBeaconBlockRoot)
+	if err := checkParentBeaconBlockRootMatch(attrs.ParentBeaconBlockRoot, envelope.ParentBeaconBlockRoot); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkParentBeaconBlockRootMatch(attrRoot, blockRoot *common.Hash) error {
```
Comment (trianglesphere) [inline op-node/rollup/derive/engine_consolidate.go]: Was this accidentally committed? I think that this is what is conflicting

Comment (protolambda) [reply op-node/rollup/derive/engine_consolidate.go]: It was a stacked PR, but the dependency merged. Will rebase and drop the old dependency commit

**Comment by protolambda** [comment]
rebased it, dropped the commit that was conflicting due to dependency PR squash change

**Pull request merged**